iModelCore/iModelPlatform/DgnCore/CodeSpec.cpp

```diff
@@ -566,7 +568,26 @@ DgnCode DgnCode::FromJson(BeJsConst value, DgnDbCR db) {
         auto specPtr = db.CodeSpecs().GetCodeSpec(specJson.asCString());
         val.m_specId = specPtr.IsValid() ? specPtr->GetCodeSpecId() : CodeSpec::GetNullCodeSpecId();
     }
-    val.m_scope = value[json_scope()].asString();
-    val.m_value = value[json_value()].asString();
+
+    // the code scope must refer to an existing element. The json may have an ElementId or FederationGuid
+    auto scopeJson = value[json_scope()];
+    auto scopeStr = scopeJson.asCString();
+
+    // Note: this is necessary since GetId64 parses strings that start with number successfully and can get confused with Guids.
+    // ElementIds in JSON must start with "0x"
+    auto validId = scopeStr && (scopeStr[0] == '0' && (scopeStr[1] == 'X' || scopeStr[1] == 'x'));

```

Comment:
You are permitting uppercase X in Ids? We forbid it elsewhere.

Reply:
That code comes from BeJsConst::GetUInt64. Should we not allow it there?

Reply:
Meh, I don't know. BeInt64Id uses BeStringUtilities::HasHexPrefix which allows uppercase too.
TypeScript only permits lowercase.

Reply:
that's why i used `scopeId.ToHexStr()` so that it will always be "correct" in the iModel.